tests/integration/conftest.py

```diff
@@ -985,12 +1036,102 @@ def assert_qtest_property(model, prop_name, exp_prop_value, promiscuous=True):
     else:
         raise AssertionError("The '{}' property not found in the provided swagger_client model!".format(prop_name))
 
+    return actual_values
+
+
+# noinspection PyUnresolvedReferences
+@pytest.helpers.register
+def assert_qtest_property(model, prop_name, exp_prop_value, promiscuous=True):
+    """Assert that a qTest swagger_client model has a property that matches the expected value. This assert will
+    intelligently search both standard qTest swagger_client model attributes as well as custom fields set by the
+    qTest admin of the qTest project under test.
+
+    Enabling 'promiscuous' mode (default) will attempt to match the property value against either the 'field_value' or
+    'field_value_name' attributes of the model. If 'promiscuous' mode is disabled then validation of the expected
+    property value will only be made against the 'field_value' attribute of the model.
+
+    Args:
+        model (object): Any model from the 'swagger_client.models' namespace.
+        prop_name (str): Target property name to use for value validation.
+        exp_prop_value (str): Expected value for the given property name.
+        promiscuous (bool): Flag for indicating whether to run assertion in 'promiscuous' mode or not. (See above)
+
+    Raises:
+        AssertionError: Property name does not exist or property value does not match expected value.
+    """
+
+    actual_values = get_qtest_property(model, prop_name)
+
     if promiscuous:
         assert exp_prop_value in actual_values
     else:
         assert actual_values[0] == exp_prop_value
 
 
+# noinspection PyUnresolvedReferences
+@pytest.helpers.register
+def assert_qtest_property_match(model, prop_name, exp_prop_regex, regex_flags=0, promiscuous=True):
```

Comment:
Turns out that extra characters can be added depending on the format of the JUnitXML. So needed to have a way to run regex pattern matching.